### PR TITLE
Add proper test setup instructions to Automated Testing documentation

### PR DESCRIPTION
**Pull request by connorshea**
This PR does a few things:

- Adds proper setup instructions for the test suite.
  - Previously, the documentation didn't mention anything about setting up the database, the Apache server, the CRM install process, or the necessity of running ChromeDriver.
  - It didn't explicitly state how to perform the API SQL dumps.
- Reorganizes the automated testing documentation to be a bit simpler and explain the setup process first, at the top of the page. It's now organized so the user can follow the instructions from top to bottom.
- Splits Docker usage into its own section.
- Move the PATH instructions to the bottom of the page under "Other tips" for simplicity.
- Various miscellaneous wording improvements.

This PR only mentions things that are currently available in at least the hotfix-7.10.x branch, no CRM PRs need to be merged before this can be used.

### CODE REVIEW

File: content/developer/automatedtesting/_index.en.adoc [hunk 1]
```diff
@@ -10,149 +10,179 @@ title: "Automated Testing"
...
+You can set up the CRM for testing with these steps:
 
+- In your terminal emulator or command prompt: `cd path/to/suitecrm/instance`
+- Install dependencies with https://getcomposer.org[Composer]: `composer install`
+- Create a MySQL database called `automated_tests`, along with a user called `automated_tests` and the password `automated_tests`. In the MySQL command line interface:
+  - `mysql -e "CREATE DATABASE automated_tests CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;"`
+  - `mysql -u root -p -e "CREATE USER 'automated_tests'@'localhost' IDENTIFIED BY 'automated_tests';"`
+  - `mysql -u root -p -e "GRANT ALL PRIVILEGES ON automated_tests.* TO 'automated_tests'@'localhost';"`
+- Configure environment variables, see environment variables section below for more information.
+- Configure your Apache server so the SuiteCRM server is available at http://localhost (or whatever instance URL you've set up in the previous step).
```
Comment (connorshea) [inline content/developer/automatedtesting/_index.en.adoc]: This part should probably be expanded upon, right now it's essentially "draw a circle, now draw the rest of the owl". I intend to do that later, but for now the documentation at least mentions the Apache server at all :)

File: content/developer/automatedtesting/_index.en.adoc [hunk 2]
```diff
@@ -10,149 +10,179 @@ title: "Automated Testing"
...
-`./vendor/bin/codecept run acceptance`
+You can set up the CRM for testing with these steps:
 
+- In your terminal emulator or command prompt: `cd path/to/suitecrm/instance`
+- Install dependencies with https://getcomposer.org[Composer]: `composer install`
+- Create a MySQL database called `automated_tests`, along with a user called `automated_tests` and the password `automated_tests`. In the MySQL command line interface:
+  - `mysql -e "CREATE DATABASE automated_tests CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;"`
+  - `mysql -u root -p -e "CREATE USER 'automated_tests'@'localhost' IDENTIFIED BY 'automated_tests';"`
+  - `mysql -u root -p -e "GRANT ALL PRIVILEGES ON automated_tests.* TO 'automated_tests'@'localhost';"`
+- Configure environment variables, see environment variables section below for more information.
```
Comment (connorshea) [inline content/developer/automatedtesting/_index.en.adoc]: Dump question, how do you link to another section in the same page with asciidoc?

**Comment by connorshea** [comment]
Some other minor notes:
- This doesn't mention the instructions for setting up ElasticSearch to be used with the test suite, I'm not sure how to do that honestly. I guess I can add a placeholder in the instructions that says "set up ElasticSearch (7.11+)"
- This doesn't mention anything about how to run an individual test vs. the whole suite.

I don't think either of these are strictly necessary for getting this merged, but I wanted to mention them before I forgot :)

**Comment by connorshea** [comment]
@pgorod could you review? :)

**Comment by pgorod** [comment]
Ah sorry, this was forgotten. I asked Dillon to review since I don't know about tests.

Anyway I would trust you on this one, so if Dillon can't pick this up in a few days I don't mind just merging

**Pull request merged**